### ISSUE TITLE
Difficult to understand the error message about rewrite hdf

### ISSUE THREAD

**Issue by samwaseda** [pyiron/pyiron_base#572]
https://github.com/pyiron/pyiron_base/blob/97e1d0a523c45720f897a5dee906e3ed9efe8449/pyiron_base/generic/hdfio.py#L210

First problem: For both `rewrite_hdf5` and `update_scripts/pyiron_base_0.3_to_0.4.py` the path is not written. Then, since I was interested in solving the problem altogether, I tried to run `pyiron_base_0.3_to_0.4.py` and got the following error message:

```python
Traceback (most recent call last):
  File "pyiron_base_0.3_to_0.4.py", line 45, in <module>
    for l in subprocess.getoutput(f"find {sys.argv[1]} -regex \".*\.h5\" -exec wc -c '{{}}' \;").split("\n"):
IndexError: list index out of range
```

I imagined that `rewrite_hdf5` is meant to be fore `job.project_hdf5.rewrite_hdf5(job_name=job.job_name)` (which also worked). I think it's fairly difficult to figure out that they have to look for `job.project_hdf5` and also find out that the job name is to be found under `job.job_name`. In my opinion the warning should explicitly say `job.project_hdf5.rewrite_hdf5(job_name=job.job_name)` and the other solution could look like something like this?

```python
import pyiron_base_update
pyiron_base_update.run()
```

**Comment by niklassiemer** [comment]
I agree, the warning is thrown from the `FileHDFio` class and not helpful in how to resolve this, especially, normally everyone will encounter this on the `job` level. I also started to work on a different route to update the data structures in #525 but did not yet come back to this.

**Comment by niklassiemer** [comment]
Fixed in #607
